src/libcommonserver/log/log.cpp

```diff
@@ -109,4 +115,393 @@ Log::Log(const log4cplus::tstring &filePath) {
+    if (!IoHelper::createDirectory(tempDirectory.parent_path(), ioError) && ioError != IoErrorDirectoryExists) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in IoHelper::createDirectory: " << Utility::formatIoError(tempDirectory.parent_path(), ioError).c_str());
+        if (ioError == IoErrorDiskFull) {
+            exitCause = ExitCauseNotEnoughDiskSpace;
+        } else {
+            exitCause = ExitCauseUnknown;
+        }
+        return ExitCodeSystemError;
+    }
+
+    if (!IoHelper::createDirectory(tempDirectory, ioError)) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in IoHelper::createDirectory: " << Utility::formatIoError(tempDirectory, ioError).c_str());
+        IoError ignoreError;
+        IoHelper::deleteDirectory(tempDirectory.parent_path(), ignoreError);
+        if (ioError == IoErrorDiskFull) {
+            exitCause = ExitCauseNotEnoughDiskSpace;
+        } else {
+            exitCause = ExitCauseUnknown;
+        }
+        return ExitCodeSystemError;
+    }
+
+    ExitCode exitCode = copyLogsTo(tempDirectory, includeOldLogs, exitCause);
+    if (exitCode != ExitCodeOk) {
+        LOG_WARN(Log::instance()->getLogger(), "Unable to copy logs to temp folder: " << exitCause);
+        IoHelper::deleteDirectory(tempDirectory.parent_path(), ioError);
+        return exitCode;
+    }
+
+    // Copy .parmsdb to temp folder
+    exitCode = copyParmsDbTo(tempDirectory / ".parms.db", exitCause);
+    if (exitCode != ExitCodeOk) {
+        LOG_WARN(Log::instance()->getLogger(), "Unable to copy .parms.db to temp folder: " << exitCause);
+        IoHelper::deleteDirectory(tempDirectory.parent_path(), ioError);
+        return exitCode;
+    }
+
+    // Generate user description file
+    exitCode = generateUserDescriptionFile(tempDirectory, exitCause);
+    if (exitCode != ExitCodeOk) {
+        LOG_WARN(Log::instance()->getLogger(), "Unable to generate user description file: " << exitCause);
+        IoHelper::deleteDirectory(tempDirectory.parent_path(), ioError);
+        return exitCode;
+    }
+
+    // compress all the files in the folder
+    exitCode = compressLogFiles(tempDirectory, exitCause, progressCallback);
+    if (exitCode != ExitCodeOk) {
+        LOG_WARN(Log::instance()->getLogger(), "Unable to compress logs: " << exitCause);
+        IoHelper::deleteDirectory(tempDirectory.parent_path(), ioError);
+        return exitCode;
+    }
+
+    // Generate the archive
+    int err = 0;
+    zip_t *archive = zip_open((logPath / "send_log_directory_temp" / archiveName).string().c_str(), ZIP_CREATE | ZIP_EXCL, &err);
+    if (err != ZIP_ER_OK) {
+        LOG_WARN(Log::instance()->getLogger(), "Error in zip_open: " << zip_strerror(archive));
+        IoHelper::deleteDirectory(tempDirectory.parent_path(), ioError);
+
+        exitCause = ExitCauseUnknown;
+        return ExitCodeUnknown;
+    }
+
+    IoHelper::DirectoryIterator dir;
+    DirectoryEntry entry;
```

Comment:
You can move this declaration where the variable is used for the first time.

Reply:
Done in commit: https://github.com/Infomaniak/desktop-kDrive/pull/100/commits/d7636060fd94c4d3a8a14f2e5366f89588dd10eb